### DISCUSSION TITLE
How do you train a TTS model with a common voice dataset?

### DISCUSSION THREAD

**Discussion by crisdesivo** [coqui-ai/TTS#1143] [General Q&A]
I found in the TTS code a formatter specifically designed for common voice, but I'm not able to make it work.
The first error I encountered was that common voice uses mp3 format but TTS uses wav. I found in the STT documentation that the recommended fix was a deepSpeech script "bin/import_cv2.py", but trying to use that brought a lot of issues that I found online without resolution, so I decided to convert the mp3 to wav myself, that was simple enough, but it is still not working, now I am getting a cleaner error during training, looks like it's trying to get a cleaner attribute by using something other than a string:
`Original Traceback (most recent call last):
  File "/usr/local/lib/python3.7/dist-packages/torch/utils/data/_utils/worker.py", line 287, in _worker_loop
    data = fetcher.fetch(index)
  File "/usr/local/lib/python3.7/dist-packages/torch/utils/data/_utils/fetch.py", line 49, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/usr/local/lib/python3.7/dist-packages/torch/utils/data/_utils/fetch.py", line 49, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/content/TTS/TTS/tts/datasets/dataset.py", line 379, in __getitem__
    return self.load_data(idx)
  File "/content/TTS/TTS/tts/datasets/dataset.py", line 236, in load_data
    add_blank=self.add_blank,
  File "/content/TTS/TTS/tts/utils/text/__init__.py", line 214, in text_to_sequence
    sequence += _symbols_to_sequence(_clean_text(text, cleaner_names))
  File "/content/TTS/TTS/tts/utils/text/__init__.py", line 252, in _clean_text
    cleaner = getattr(cleaners, name)
TypeError: getattr(): attribute name must be string`

This is my code:
```
import os
import sys
print(os.getcwd())
sys.path.append('/content/TTS')

from TTS.trainer import Trainer, TrainingArgs

from TTS.tts.configs.glow_tts_config import GlowTTSConfig

from TTS.tts.configs.shared_configs import BaseDatasetConfig
from TTS.tts.datasets import load_tts_samples
from TTS.tts.models.glow_tts import GlowTTS
from TTS.utils.audio import AudioProcessor

output_path = "/content"

dataset_config = BaseDatasetConfig(
    meta_file_train="/content/train.tsv",
    path="/content",
    name="common_voice"
)

config = GlowTTSConfig(
    batch_size=32,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=-1,
    epochs=100,
    use_phonemes=False,
    print_step=25,
    print_eval=False,
    mixed_precision=True,
    output_path=output_path,
    datasets=[dataset_config],
)
configDict = config.audio.to_dict()
configDict["sample_rate"] = 48000
print(configDict)
ap = AudioProcessor(**configDict)


train_samples, eval_samples = load_tts_samples(dataset_config, eval_split=True)

model = GlowTTS(config, speaker_manager=None)

trainer = Trainer(
    TrainingArgs(),
    config,
    output_path,
    model=model,
    train_samples=train_samples,
    eval_samples=eval_samples,
    training_assets={"audio_processor": ap},  # assets are objetcs used by the models but not class members.
)

trainer.fit()
```

I'm doing this on a colab notebook but it also doesn't work on my ubuntu 20.04 with python 3.8

Does anyone have any insight on this?

**Comment by mattkanwisher** [comment]
Rather old post, but I think the answer is, you're not supposed to. STT (Speech to text), repo has a python script for this. Speech to text uses many speakers. TTS (text to speech) seems to be trained on a single speaker so you need like 24 hours of audio of a single speaker

**Comment by erogol** [comment]
You just need to write or use the right formatter. Check TTS/tts/datasets/formatters.py

**Reply by ianujkr** [reply]
Can u post the correct formatters.py for the mp3 file or suggest any solution. I am also using the common voice dataset for training and I am also facing the same issue.
